storage/imxrt-flash/flashdrv.c

```diff
@@ -119,7 +119,68 @@ static ssize_t bufferSync(flash_context_t *ctx, uint32_t dstAddr, int isLast)
 }
 
 
-ssize_t flash_bufferedPagesWrite(flash_context_t *ctx, uint32_t dstAddr, const void *srcPtr, size_t size)
+ssize_t flash_bufferedRead(flash_context_t *ctx, uint32_t srcAddr, void *dstPtr, size_t size)
```

Comment:
It might be possible to refactor this code to handle each case in sequence manner instead of handling every permutation separately